source/types/ut_assert_result_base.tps

```diff
@@ -0,0 +1,13 @@
+create or replace type ut_assert_result_base as object
+(
+  result   integer(1),
+  message  varchar2(4000 char),
+  expected varchar2(4000 char),
+  actual   varchar2(4000 char),
+  end_time timestamp with time zone,
+  not instantiable member procedure add_element( self in out nocopy ut_assert_result_base, an_assertion ut_assert_result_base ),
```

Comment:
I see your point of making base class for an assertion result but I don't think "expected" and "actual" properties should be there, I think they should be moved to the subclass like "ut_compart_assert", it assert's may theck error raised, or something else that can't be represented natively by "expected", "actual" values.